File: demo/src/main/java/ua/at/tsvetkov/demo/AppTaoLogDemo.java
package ua.at.tsvetkov.demo;

import android.app.Application;

import ua.at.tsvetkov.util.logger.LogComponents;
import ua.at.tsvetkov.util.logger.Log;

/**
 * Created by lordtao on 06.12.2017.
 */

public class AppTaoLogDemo extends Application {

   @Override
   public void onCreate() {
      super.onCreate();

      if (BuildConfig.DEBUG){
         Log.setStamp(BuildConfig.GIT_SHA);
         LogComponents.enableComponentsChangesLogging(this);
      } else {
         Log.setDisabled();
      }

   }

}

File: demo/src/main/java/ua/at/tsvetkov/demo/AppTaoLogDemo.kt
package ua.at.tsvetkov.demo

import android.app.Application
import ua.at.tsvetkov.util.BuildConfig
import ua.at.tsvetkov.util.logger.Log.setDisabled
import ua.at.tsvetkov.util.logger.LogComponents.enableComponentsChangesLogging

/**
 * Created by lordtao on 06.12.2017.
 */
class AppTaoLogDemo : Application() {

    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            enableComponentsChangesLogging(this)
        } else {
            setDisabled()
        }
    }

}